#!/usr/bin/env python
PACKAGE = "potbot_lib"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("reset_path_index",                     bool_t, 1, "A Boolean parameter", True)
gen.add("distance_change_to_pose_alignment",    double_t, 0, "A double parameter", 0.6, 0.01, 1)
gen.add("stop_margin_angle",                    double_t, 0, "A double parameter", 0.1, 0, 0.2)
gen.add("stop_margin_distance",                 double_t, 0, "A double parameter", 0.03, 0, 0.3)
gen.add("max_linear_velocity",          double_t, 0, "A double parameter", 0.5, 0, 2)
gen.add("min_linear_velocity",          double_t, 0, "A double parameter", -0.5, -2, 0)
gen.add("max_angular_velocity",         double_t, 0, "A double parameter", 1, 0, 3)
gen.add("min_angular_velocity",         double_t, 0, "A double parameter", -1, -3, 0)
gen.add("time_increment",           double_t, 0, "A double parameter", 0.1, 0.001, 1.0)
gen.add("time_end",           double_t, 0, "A double parameter", 1, 0.01, 10.0)
gen.add("linear_velocity_increment",double_t, 0, "A double parameter", 0.05, 0.001, 0.3)
gen.add("angular_velocity_increment",double_t, 0, "A double parameter", 0.1, 0.001, 1.0)
gen.add("learning_rate",double_t, 0, "A double parameter", 0.01, 0.0, 1)

exit(gen.generate(PACKAGE, "potbot_lib", "DWA"))